haskell/nixpkgs.bzl

```diff
@@ -242,6 +249,19 @@ def haskell_register_ghc_nixpkgs(
     _ghc_nixpkgs_toolchain(name = toolchain_repo_name)
     native.register_toolchains("@{}//:toolchain".format(toolchain_repo_name))
 
+    # Unix tools toolchain required for Cabal packages
+    unix_nixpkgs_kwargs = dict(
+        nixopts = nixopts,
+        repositories = repositories,
+        repository = repository,
+    )
+    if unix_attributes != None:
+        unix_nixpkgs_kwargs["packages"] = unix_attributes
+    unix_nixpkgs(
```

Comment:
https://github.com/tweag/rules_haskell/pull/1117/files#diff-85408e6f9e4818174d8baf3df5946e07R456 guards the call to `unix_configure` depending on whether the toolchain already exists. Is there a reason why it isn't the case here too or is it just an overlook?

Reply:
Yes, there's a reason. In this case `_ghc_nixpkgs_toolchain` is created with a hard-coded name, so this repository macro can only be invoked once. Therefore there's no need for this check.

However in `haskell_register_ghc_bindists` the workspace name contains the GHC version number, so it could be called multiple times for different GHC versions. In that case we need to check to avoid name collisions.